jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/ConfigurationPropertyValidator.java

```diff
@@ -128,5 +137,58 @@ public static ImageReference getGeneratedTargetDockerTag(
     }
   }
 
+  /**
+   * Parses a string in the form of "key1=value1,key2=value2,..." into a map.
+   *
+   * @param property the map string to parse, with entries separated by "," and key-value pairs
+   *     separated by "="
+   * @return the map of parsed values
+   */
+  public static Map<String, String> parseMapProperty(String property) {
+    Map<String, String> result = new HashMap<>();
+
+    // Split on non-escaped commas
+    List<String> entries = parseListProperty(property);
+    for (String entry : entries) {
+      Matcher matcher = environmentPattern.matcher(entry);
+      if (!matcher.matches()) {
+        throw new IllegalArgumentException("'" + entry + "' is not a valid key-value pair");
+      }
+      result.put(matcher.group("name"), matcher.group("value"));
+    }
+    return result;
+  }
+
+  /**
+   * Parses a comma-separated string into a list. Ignores commas escaped with "\".
+   *
+   * @param property the comma-separated string
+   * @return the list of parsed values
+   */
+  public static List<String> parseListProperty(String property) {
+    List<String> items = new ArrayList<>();
+    boolean shouldEscape = false;
+    int startIndex = 0;
+    int endIndex = 0;
+    for (; endIndex < property.length(); endIndex++) {
+      // If previously encountered a backslash, skip this character
+      if (shouldEscape) {
+        shouldEscape = false;
+        continue;
+      }
+
+      if (property.charAt(endIndex) == ',') {
+        // Split on non-escaped comma
+        items.add(property.substring(startIndex, endIndex));
+        startIndex = endIndex + 1;
+      } else if (property.charAt(endIndex) == '\\') {
+        // Found a backslash, ignore next character
+        shouldEscape = true;
```

Comment:
Probably we can just do `endIndex++` here and remove the `shouldEscape` variable?